File: slack-poster/src/test/java/dev/number6/slackposter/SlackServiceTest.java
package dev.number6.slackposter;

import dev.number6.slackposter.port.SlackPort;
import dev.number6.slackposter.model.ChannelSummaryImageBuilder;
import com.amazonaws.services.lambda.runtime.LambdaLogger;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.junit.jupiter.MockitoExtension;

import java.io.IOException;

import static org.mockito.ArgumentMatchers.any;
import static org.mockito.Mockito.*;

@ExtendWith(MockitoExtension.class)
class SlackServiceTest {

    @Mock
    LambdaLogger logger;
    @Mock
    SlackPort slackPort;
    @InjectMocks
    SlackService testee;

    @Test
    void postToSlackIfLastUpdate() throws IOException {

        ChannelSummaryImage image = ChannelSummaryImageBuilder.finalImage().build();
        testee.handleNewImage(image, logger);
        verify(slackPort, times(1)).postMessageToChannel(any(), any());
    }

    @Test
    void doNotPostToSlackIfNotLastUpdate() {

        ChannelSummaryImage image = ChannelSummaryImageBuilder.notFinalImage().build();
        testee.handleNewImage(image, logger);
        verifyZeroInteractions(slackPort);
    }


}

File: slack-poster/src/test/kotlin/dev/number6/slackposter/SlackServiceTest.kt
package dev.number6.slackposter

import com.amazonaws.services.lambda.runtime.LambdaLogger
import dev.number6.slackposter.model.ChannelSummaryImageBuilder
import dev.number6.slackposter.port.SlackPosterPort
import io.mockk.Called
import io.mockk.junit5.MockKExtension
import io.mockk.mockk
import io.mockk.verify
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import java.io.IOException

@ExtendWith(MockKExtension::class)
internal class SlackServiceTest {

    private val logger: LambdaLogger = mockk(relaxUnitFun = true)

    private val slackPosterPort: SlackPosterPort = mockk(relaxUnitFun = true)

    @Test
    @Throws(IOException::class)
    fun postToSlackIfLastUpdate() {
        val testee = SlackService(slackPosterPort)
        val image: ChannelSummaryImage = ChannelSummaryImageBuilder.finalImage().build()
        testee.handleNewImage(image, logger)
        verify { slackPosterPort.postMessageToChannel(any(), any()) }
    }

    @Test
    fun doNotPostToSlackIfNotLastUpdate() {
        val testee = SlackService(slackPosterPort)
        val image: ChannelSummaryImage = ChannelSummaryImageBuilder.notFinalImage().build()
        testee.handleNewImage(image, logger)
        verify { slackPosterPort wasNot Called }
    }
}